services/supervisord/supervisord.go

```diff
@@ -393,9 +398,20 @@ func (s *Service) reload(name string) error {
 
 // marshalConfig marshals supervisord program configuration.
 func (s *Service) marshalConfig(tmpl *template.Template, settings *models.Settings) ([]byte, error) {
+	retentionMonths := int(math.Ceil(settings.DataRetention.Hours() / 24 / 30))
+	if retentionMonths <= 0 {
+		retentionMonths = 1
+	}
 	templateParams := map[string]interface{}{
-		"DataRetentionHours": int(settings.DataRetention.Hours()),
-		"DataRetentionDays":  int(settings.DataRetention.Hours() / 24),
+		"DataRetentionHours":  int(settings.DataRetention.Hours()),
+		"DataRetentionDays":   int(settings.DataRetention.Hours() / 24),
+		"DataRetentionMonths": retentionMonths,
+		"IsVMEnabled":         s.vmParams.Enabled,
+		"VMAlertParams":       s.vmParams.VMAlertFlags,
+		"VMDBCacheEnable":     !settings.VictoriaMetrics.CacheEnable,
```

Comment:
Why `!`?

Reply:
Vmdb has setting `-search.disableCache`, so boolean param CacheEnable must be inverted.

Reply:
but you pass it as `VMDBCacheEnable`, not as `VMDBCacheDisable`

Reply:
Yep, it's misleading, thx!

Reply:
changed